lib/commons/text/is-icon-ligature.js

```diff
@@ -0,0 +1,189 @@
+	 *
+	 * But if the entire string "File" produced an image which had at least
+	 * a 10% difference in pixels, we would know that the string was replaced
+	 * by a ligature:
+	 *
+	 * ┌─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┐
+	 * │ │█│█│█│█│█│█│█│█│█│█│ │ │ │ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │ │ │ │ │ │ │ │█│█│ │ │ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │█│█│█│█│█│█│ │█│ │█│ │ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │ │ │ │ │ │ │ │█│█│█│█│ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │█│█│█│█│█│█│ │ │ │ │█│ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │ │ │ │ │ │ │ │ │ │ │█│ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │█│█│█│█│█│█│█│█│█│ │█│ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│ │ │ │ │ │ │ │ │ │ │ │█│ │
+	 * ├─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┼─┤
+	 * │ │█│█│█│█│█│█│█│█│█│█│█│█│█│ │
+	 * └─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┘
+	 */
+	const nodeValue = textVNode.actualNode.nodeValue;
+
+	// text with unicode or non-bmp letters cannot be ligature icons
+	if (
+		!text.sanitize(nodeValue) ||
+		text.hasUnicode(nodeValue, { emoji: true, nonBmp: true })
+	) {
+		return false;
+	}
+
+	if (!axe._cache.get('context')) {
+		axe._cache.set(
+			'context',
+			document.createElement('canvas').getContext('2d')
+		);
+	}
+	const context = axe._cache.get('context');
+	const canvas = context.canvas;
+
+	// keep track of each font encountered and the number of times it shows up
+	// as a ligature.
+	if (!axe._cache.get('fonts')) {
+		axe._cache.set('fonts', {});
+	}
+	const fonts = axe._cache.get('fonts');
+
+	const style = window.getComputedStyle(textVNode.parent.actualNode);
+	const fontFamily = style.getPropertyValue('font-family');
+
+	if (!fonts[fontFamily]) {
+		fonts[fontFamily] = {
+			occurances: 0,
+			numLigatures: 0
+		};
+	}
+	const font = fonts[fontFamily];
+
+	// improve the performance by only comparing the image data of a font
+	// a certain number of times
+	if (font.occurances >= occuranceThreshold) {
+		// if the font has always been a ligature assume it's a ligature font
+		if (font.numLigatures / font.occurances === 1) {
+			return true;
+		}
+		// inversely, if it's never been a ligature assume it's not a ligature font
+		else if (font.numLigatures === 0) {
+			return false;
+		}
+
+		// we could theoretically get into an odd middle ground scenario in which
+		// the font family is being used as normal text sometimes and as icons
+		// other times. in these cases we would need to always check the text
+		// to know if it's an icon or not
+	}
+	font.occurances++;
+
+	// 30px was chosen to account for common ligatures in normal fonts
+	// such as fi, ff, ffi. If a ligature would add a single column of
+	// pixels to a 30x30 grid, it would not meet the statistical significance
+	// threshold of 10% (30x30 = 900; 30/900 = 3.333%). this also allows for
+	// more than 1 column differences (60/900 = 6.666%) and things like
+	// extending the top of the f in the fi ligature.
```

Comment:
common ligatures should not get a pass on this rule. In these cases, the visible text (w/ `fi` ligature) "final" should match an `aria-label`. Only "final" converted to an icon would get a pass since the label/accName could be anything associated visually with the icon.